### ISSUE TITLE
issues with devel version of liquid-dsp

### ISSUE THREAD

**Issue by andreicociuba** [quiet/quiet#8]
I have installed (and triple-checked) that the DEVEL version of liquid-dsp is installed on my ubuntu machine, but apparently some or other of the required functions do not exist in the current version, so the bootstrap script fails with a message to download the devel version, which i did to begin with.

my guess is that some or other functions have been deprecated even in the devel version of liquid dsp, but my programming skills are not good enough on linux and gcc to actually be able to know where to look specifically.

i am glad to help if you could tell me what kind of additional information to give you.

**Comment by andreicociuba** [comment]
CMakeError.log contents


Determining if the function flexframesync_set_header_len,flexframegen_set_header_len,ofdmflexframesync_set_header_len,ofdmflexframegen_set_hea$
Change Dir: /home/an3/quiet-master/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_15661/fast"
/usr/bin/make -f CMakeFiles/cmTC_15661.dir/build.make CMakeFiles/cmTC_15661.dir/build
make[1]: Entering directory '/home/an3/quiet-master/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_15661.dir/CheckFunctionExists.c.o
/usr/bin/cc    -O2 -std=c99 -Wpedantic -Werror -Wall -D_XOPEN_SOURCE=700 -DCHECK_FUNCTION_EXISTS=flexframesync_set_header_len,flexframegen_set$
/usr/share/cmake-3.5/Modules/CheckFunctionExists.c: In function ‘main’:
<command-line>:0:51: error: left-hand operand of comma expression has no effect [-Werror=unused-value]
/usr/share/cmake-3.5/Modules/CheckFunctionExists.c:14:3: note: in expansion of macro ‘CHECK_FUNCTION_EXISTS’
   CHECK_FUNCTION_EXISTS();
   ^
<command-line>:0:79: error: left-hand operand of comma expression has no effect [-Werror=unused-value]
/usr/share/cmake-3.5/Modules/CheckFunctionExists.c:14:3: note: in expansion of macro ‘CHECK_FUNCTION_EXISTS’
   CHECK_FUNCTION_EXISTS();
   ^
<command-line>:0:112: error: left-hand operand of comma expression has no effect [-Werror=unused-value]
/usr/share/cmake-3.5/Modules/CheckFunctionExists.c:14:3: note: in expansion of macro ‘CHECK_FUNCTION_EXISTS’
   CHECK_FUNCTION_EXISTS();
   ^
<command-line>:0:144: error: left-hand operand of comma expression has no effect [-Werror=unused-value]
/usr/share/cmake-3.5/Modules/CheckFunctionExists.c:14:3: note: in expansion of macro ‘CHECK_FUNCTION_EXISTS’
   CHECK_FUNCTION_EXISTS();
   ^
<command-line>:0:177: error: left-hand operand of comma expression has no effect [-Werror=unused-value]
/usr/share/cmake-3.5/Modules/CheckFunctionExists.c:14:3: note: in expansion of macro ‘CHECK_FUNCTION_EXISTS’
   CHECK_FUNCTION_EXISTS();
   ^
cc1: all warnings being treated as errors
CMakeFiles/cmTC_15661.dir/build.make:65: recipe for target 'CMakeFiles/cmTC_15661.dir/CheckFunctionExists.c.o' failed
make[1]: *** [CMakeFiles/cmTC_15661.dir/CheckFunctionExists.c.o] Error 1
make[1]: Leaving directory '/home/an3/quiet-master/build/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_15661/fast' failed
make: *** [cmTC_15661/fast] Error 2

**Comment by brian-armstrong** [comment]
Hi Andrei,

I have to admit, I'm not even close to being an expert at debugging CMake failures, and this one looks pretty opaque.

Just to clarify about these functions, this branch is my development branch of my fork of liquid dsp. The functions are actually all very new and should be present. Running `$ grep ofdmflexframegen_set_header_len /usr/local/include/liquid/liquid.h` should confirm the presence of one of these new functions (your path may be different - this is the path for user-installed headers on OSX, and I can't remember whether Ubuntu would be the same)

Looking at the output, I'm thinking that gcc with -Wpedantic might be erroring on something that clang on -Wpedantic does not, but I'd need to check to be sure.

**Comment by andreicociuba** [comment]
running this bash script to check for the function headers:
#!/bin/bash
ARRAY=("flexframesync_set_header_len" "flexframegen_set_header_len" "ofdmflexframesync_set_header_len" "ofdmflexframegen_set_header_len" "resa$

for i in "${ARRAY[@]}"
do
echo
echo looking for $i
grep $i /usr/local/include/liquid/liquid.h
echo if there is no output before this, it hasnt been found
done


yields the following result:

looking for flexframesync_set_header_len
void flexframesync_set_header_len(flexframesync _q,
void ofdmflexframesync_set_header_len(ofdmflexframesync _q,
if there is no output before this, it hasnt been found

looking for flexframegen_set_header_len
void flexframegen_set_header_len(flexframegen _q, unsigned int _len);
void ofdmflexframegen_set_header_len(ofdmflexframegen _q,
if there is no output before this, it hasnt been found

looking for ofdmflexframesync_set_header_len
void ofdmflexframesync_set_header_len(ofdmflexframesync _q,
if there is no output before this, it hasnt been found

looking for ofdmflexframegen_set_header_len
void ofdmflexframegen_set_header_len(ofdmflexframegen _q,
if there is no output before this, it hasnt been found

looking for resamp_rrrf_execute_output_block
if there is no output before this, it hasnt been found

looking for liquid_pack_array_block
void liquid_pack_array_block(unsigned char * _dest,
if there is no output before this, it hasnt been found


meaning that the function resamp_rrrf_execute_output_block ISNT there.

either that, and/or the fact that the cmake WARNINGS are treated as errors might cause the entire check of the presence of the required functions to fail.....

**Comment by brian-armstrong** [comment]
I think the reason you might not be seeing resamp_rrrf_execute_output_block is because of how liquid creates multiple types with macros. In this case, you'd want to see if you can find `RESAMP(_execute_output_block)` which will be expanded into `resamp_rrrf_execute_output_block` by line 3022 https://github.com/quiet/liquid-dsp/blob/devel/include/liquid.h#L3022

I bet you might be on to something with those warnings. If you remove `-Werror` from this line in CMakeLists.txt, does that change anything? https://github.com/quiet/quiet/blob/master/CMakeLists.txt#L28

**Comment by andreicociuba** [comment]
Yes, i did find RESAMP(_execute_output_block)
no, removing -Werror from line 28 of CMakeLists.txt didnt do any difference

now i am investigating online what the default location is for user installed header files, maybe they have been placed on installation in the wrong directory. yes i did run sudo  ldconfig....

**Comment by andreicociuba** [comment]
according to this answer http://stackoverflow.com/questions/11132153/location-of-c-header-files-in-debian, they should be in the right location. assuming that ubuntu stuck to the debian defaults, that is, on this particular issue....

**Comment by brian-armstrong** [comment]
I'm sorry to say I don't know anything else you could try.

I'm going to spin up an Ubuntu VM today and have a try myself. Portability is always tricky but I would like to make sure Ubuntu users have an easy time installing this library.

Thanks for your feedback so far and I'll let you know soon what I find out.

**Comment by brian-armstrong** [comment]
Hi Andrei,

I managed to reproduce the error you saw in Ubuntu and have created a fix for it in my installation. I went ahead and pushed that fix to the master branch, so either pull the latest work or delete your copy and reclone. I think this should get it working for you.

I wasn't able to get the install working with -Werror on, and I'm not sure why. I definitely would like to add that flag back in but for now it seems I'll just have to turn it off.
